File: mvpreaderjava/src/main/java/com/example/mvpreaderjava/modle/exception/ApiException.java
package com.example.mvpreaderjava.modle.exception;

/**
 * Created by sw116 on 2018/4/25.
 * 本地api错误
 */
public class ApiException extends RuntimeException {
    public ApiException(){
        super();
    }

    public ApiException(String msg){
        super(msg);
    }
}

File: app/src/main/java/com/savion/mvpreader/model/exception/ApiException.kt
package com.savion.mvpreader.model.exception

import java.lang.Exception


/**
 * Created by savion on 2018/4/10.
 */
class ApiException : Exception {

    private var code: Int = 0

    constructor() : super("unknown api exception occur")

    constructor(msg: String) : super(msg)

    constructor(code: Int, msg: String) : super(msg) {
        this.code = code
    }


}